### PR TITLE
Remove -b option from restore-backup.  

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Removing the -b option from restore-backup, updated tests accordingly.  --constraints and --build-agent were also removed as they were specific to the -b option.  Added new restore-backup unit tests.

## QA steps

juju restore-backup -b should now return "ERROR flag provided but not defined: -b".

## Documentation changes

Yes - please review updates to juju restore-backup command help which are part of this pr

## Bug reference

n/a

### CODE REVIEW

**Comment by hmlanigan** [comment]
[Remove -b option from backup-restore in 2.4](https://trello.com/c/huIlyoXD/247-remove-b-option-from-backup-restore-in-24)

File: cmd/juju/backups/restore.go [hunk 1]
```diff
@@ -78,29 +44,18 @@ type RestoreAPI interface {
...
-
-The given constraints will be used to choose the new instance.
+Restores a backup that was previously created with "juju create-backup",
+returning an existing controller to a previous state.
 
 If the provided state cannot be restored, this command will fail with
-an appropriate message.  For instance, if the existing bootstrap
-instance is already running then the command will fail with a message
-to that effect.
+an appropriate message.
```
Comment (mitechie) [inline cmd/juju/backups/restore.go]: s/an appropriate message/an explanation

File: cmd/juju/backups/restore.go [hunk 2]
```diff
@@ -78,29 +44,18 @@ type RestoreAPI interface {
...
-Restores a backup that was previously created with "juju create-backup".
-
-This command creates a new controller and arranges for it to replace
-the previous controller for a model.  It does *not* restore
-an existing server to a previous state, but instead creates a new server
-with equivalent state.  As part of restore, all known instances are
-configured to treat the new controller as their master.
-
-The given constraints will be used to choose the new instance.
+Restores a backup that was previously created with "juju create-backup",
```
Comment (mitechie) [inline cmd/juju/backups/restore.go]: Restores the Juju controller state database from a backup that was previously created with "juju create-backup".

Comment (mitechie) [reply cmd/juju/backups/restore.go]: Not sure on this. I feel like it should be clear this only is restoring the DB vs more aspects of Juju and controllers.

Comment (hmlanigan) [reply cmd/juju/backups/restore.go]: i'm on the fence with specifically calling out controller.  backup and restore only works if the model is controller, however it doesn't assume the controller model is being used either.  perhaps we're trying leave the door open for later allowing for a different granularity of backups?  or is left over from juju 1.x?

Comment (hmlanigan) [reply cmd/juju/backups/restore.go]: Perhaps a paragraph to say that existing units will not be removed, nor will new ones be created to fit the database being restored should be added?  With more detail.

File: cmd/juju/backups/restore.go [hunk 3]
```diff
@@ -78,29 +44,18 @@ type RestoreAPI interface {
...
 `
 
-var BootstrapFunc = bootstrap.Bootstrap
-
 // Info returns the content for --help.
 func (c *restoreCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "restore-backup",
-		Purpose: "Restore from a backup archive to a new controller.",
+		Purpose: "Restore from a backup archive to the existing controller",
```
Comment (hmlanigan) [inline cmd/juju/backups/restore.go]: Update the purpose once wording on restoreDoc is finalized.

File: cmd/juju/backups/restore.go (hunk 2 above)
Comment (mitechie) [reply cmd/juju/backups/restore.go]: The command isn't limited to only folks that are admins and folks may try to run it, get errors, etc. So the help has to be clear enough to everyone. This is why I think being explicit about what it's doing it best. I'm +1 on mentioning that units won't be removed/updated/etc. I do think that saying that restore is only updating the state database is a helpful clarification.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**